pkg/kubelet/dockertools/manager.go

```diff
@@ -280,8 +280,15 @@ func (dm *DockerManager) inspectContainer(dockerID, containerName, tPath string)
 		result.status.State.Running = &api.ContainerStateRunning{
 			StartedAt: util.NewTime(inspectResult.State.StartedAt),
 		}
-		if containerName == PodInfraContainerName && inspectResult.NetworkSettings != nil {
-			result.ip = inspectResult.NetworkSettings.IPAddress
+		if dm.networkPlugin.Name() != network.DefaultPluginName && containerName == PodInfraContainerName {
+			netStatus, err := dm.networkPlugin.Status(pod.Namespace, pod.Name, kubeletTypes.DockerID(dockerID))
+			if err == nil {
+				result.ip = netStatus.IPv4.String()
```

Comment:
I wasn't pushing to get those changes in this PR; my point was if NetworkStatus could be organized differently now to favor these later it would be helpful. For this PR I think we should add `ipv4.valid` and `ipv6.valid`, and if possible arrays of v4 and v6 addresses.